odoo/tools/image.py

```diff
@@ -65,16 +65,19 @@ def __init__(self, base64_source, verify_resolution=True):
         :rtype: ImageProcess
 
         :raise: ValueError if `verify_resolution` is True and the image is too large
-        :raise: UserError if the base64 is incorrect or the image can't be identified by PIL
+        :raise: UserError if the image can't be identified by PIL
         """
-        self.base64_source = base64_source or False
+        self.source = source or False
         self.operationsCount = 0
 
-        if not base64_source or base64_source[:1] in (b'P', 'P'):
+        if not source or source[:1] == b'<':
```

Comment:
Ok to change `P` into `<`, but we probably still need "in tuple" here, at least `source` is still documented to allow either string or bytes. (if that is no longer the case the param doc should be updated)

Reply:
source shoudl always be a binary; it's weird to pass images as strings, and I couldn't find a use case in my test. I will keep it as it is; if it crashes somewhere, we will fix the original type (it's not normal to have images as strings).
I removed the comment.

Reply:
This was added for SVG images, that are XML text.

Reply:
SVG images should be binary, not text, they are stored in the same Image field than PNG images, so they should use the same type no?

Reply:
If I don't remember bad, it was when uploading the SVG through the image widget when I got a string here, got through the JSON-RPC.